### PR TITLE
Fix genes_biodomains transform to drop duplicate rows 

### PR DESCRIPTION
**Pull request by jaclynbeck-sage**
Per [AG-1384](https://sagebionetworks.jira.com/browse/AG-1384), Ensembl IDs that map to multiple HGNC symbols cause the `genes_biodomains` output to have rows with duplicate data for that Ensembl ID once the HGNC symbol is dropped. This PR fixes that and drops duplicated rows. 

I also added test coverage for this by adding a duplicate row with a different HGNC symbol in the `biodomains_test_input_bad_but_should_pass` file. 

The `gene_info` transform already de-duplicates the data and already has coverage for duplicate biodomain/Ensembl ID combinations. `go_terms` are dropped in this transform so the `gene_info` transform is not affected by this issue. 

I also updated the biodomains test input files to include extra columns that were introduced in the new biodomains file. 

[AG-1384]: https://sagebionetworks.jira.com/browse/AG-1384?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: tests/test_assets/genes_biodomains/input/biodomains_test_input.csv [hunk 1]
```diff
@@ -1,37 +1,37 @@
...
-Proteostasis,Pr,Proteostasis [Pr],#c8b269,GO:0005783,endoplasmic reticulum,ENSG00000161011
-Proteostasis,Pr,Proteostasis [Pr],#c8b269,GO:0005796,Golgi lumen,ENSG00000188157
-Proteostasis,Pr,Proteostasis [Pr],#c8b269,GO:0042254,ribosome biogenesis,ENSG00000290146
-Proteostasis,Pr,Proteostasis [Pr],#c8b269,GO:0070475,rRNA base methylation,ENSG00000290146
-Proteostasis,Pr,Proteostasis [Pr],#c8b269,GO:0070037,rRNA (pseudouridine) methyltransferase activity,ENSG00000290146
-Proteostasis,Pr,Proteostasis [Pr],#c8b269,GO:0031625,ubiquitin protein ligase binding,ENSG00000161011
-Proteostasis,Pr,Proteostasis [Pr],#c8b269,GO:0043130,ubiquitin binding,ENSG00000161011
-Proteostasis,Pr,Proteostasis [Pr],#c8b269,GO:0016235,aggresome,ENSG00000161011
-Proteostasis,Pr,Proteostasis [Pr],#c8b269,GO:0016235,aggresome,ENSG00000000938
+biodomain,abbr,label,color,go_id,go_terms,n_symbol,symbol,ensembl_gene_id
```
Comment (jaclynbeck-sage) [inline tests/test_assets/genes_biodomains/input/biodomains_test_input.csv:1]: This file has no new test cases and is unchanged except for adding 2 new columns from the new biodomains file.

File: tests/test_assets/genes_biodomains/input/biodomains_test_input_bad_but_should_pass.csv [hunk 2]
```diff
@@ -1,8 +1,9 @@
...
-Autophagy,Au,Autophagy [Au],,GO:0000423,mitophagy,ENSG00000161011
-Autophagy,Au,Autophagy [Au],#9931fd,,aggrephagy,ENSG00000161011
-Autophagy,Au,Autophagy [Au],#9931fd,GO:0038096,,ENSG00000000938
-Autophagy,Au,Autophagy [Au],#9931fd,GO:0050764,regulation of phagocytosis,
\ No newline at end of file
+biodomain,abbr,label,color,go_id,go_terms,n_symbol,symbol,ensembl_gene_id
+,Au,Autophagy [Au],#9931fd,GO:0006914,autophagy,152,SQSTM1,ENSG00000161011
+Autophagy,,Autophagy [Au],#9931fd,GO:0016236,macroautophagy,71,SQSTM1,ENSG00000161011
+Autophagy,Au,,#9931fd,GO:0000422,autophagy of mitochondrion,43,SQSTM1,ENSG00000161011
+Autophagy,Au,,#9931fd,GO:0000422,autophagy of mitochondrion,43,DUPLICATE,ENSG00000161011
```
Comment (jaclynbeck-sage) [inline tests/test_assets/genes_biodomains/input/biodomains_test_input_bad_but_should_pass.csv:5]: Added duplicate row. The other lines in this file are unchanged except adding the extra 2 columns.

File: tests/test_assets/genes_biodomains/input/biodomains_test_input_bad_should_fail.csv [hunk 3]
```diff
@@ -1,10 +1,6 @@
...
-Autophagy,Au,Autophagy [Au],#9931fd,GO:0016236,,ENSG00000161011
-Autophagy,Au,Autophagy [Au],#9931fd,GO:0000422,autophagy of mitochondrion,
-,,,,,,
-,,,,,,
-,,,,,,
-,,,,,,
-,,,,,,
-,,,,,,
\ No newline at end of file
+biodomain,abbr,label,color,go_id,go_terms,n_symbol,symbol,ensembl_gene_id
```
Comment (jaclynbeck-sage) [inline tests/test_assets/genes_biodomains/input/biodomains_test_input_bad_should_fail.csv:1]: This file has no new test cases and is unchanged except for adding 2 new columns from the new biodomains file.

**Review by JessterB — APPROVED** [review]
lgtm!
